server/common/app_config.py

```diff
@@ -438,12 +438,24 @@ def handle_app(self, context):
         self.check_attr("app__csp_directives", (type(None), dict))
 
         if self.app__port:
-            if not is_port_available(self.app__host, self.app__port):
+            try:
+                if not is_port_available(self.app__host, self.app__port):
+                    raise ConfigurationError(
+                        f"The port selected {self.app__port} is in use, please configure an open port."
+                    )
+            except OverflowError:
+                raise ConfigurationError(f"Invalid port: {self.app__port}")
+        else:
+            try:
+                default_server_port = int(os.environ.get("CXG_SERVER_PORT", DEFAULT_SERVER_PORT))
+            except ValueError:
                 raise ConfigurationError(
-                    f"The port selected {self.app__port} is in use, please configure an open port."
+                    "Invalid port from envvar CXG_SERVER_PORT: " + os.environ.get("CXG_SERVER_PORT")
```

Comment:
not sure `envvar` is a widely known term - I have not seen it before.  Recommend the full term, `environment variable` ...